src/AutoRest.CSharp/Mgmt/Generation/ResourceContainerWriter.cs

```diff
@@ -1,63 +1,636 @@
+            var parent = "parent";
+            _writer.WriteXmlDocumentationParameter(parent, "The resource representing the parent resource.");
+            using (_writer.Scope($"internal {typeOfThis}({typeof(ResourceOperationsBase)} {parent}) : base({parent})"))
+            {
+                _writer.Line($"{ClientDiagnosticsField} = new {typeof(ClientDiagnostics)}(ClientOptions);");
+                // todo: after a shared pipeline field is implemented in the base class, replace this with that
+                _writer.Line($"{PipelineField} = {typeof(ManagementPipelineBuilder)}.Build(Credential, BaseUri, ClientOptions);");
+            }
+        }
+
+        private void WriteFields()
+        {
+            _writer.Line();
+            _writer.Line($"private readonly {typeof(ClientDiagnostics)} {ClientDiagnosticsField};");
+            _writer.Line($"private readonly {typeof(HttpPipeline)} {PipelineField};");
+
+            _writer.Line();
+            _writer.WriteXmlDocumentationSummary($"Represents the REST operations.");
+            // subscriptionId might not always be needed. For example `RestOperations` does not have it.
+            var subIdIfNeeded = _restClient.Parameters.FirstOrDefault()?.Name == "subscriptionId" ? ", Id.SubscriptionId" : "";
+            _writer.Line($"private {_restClient.Type} {OperationsField} => new {_restClient.Type}({ClientDiagnosticsField}, {PipelineField}{subIdIfNeeded});");
+        }
+
+        private void WriteIdProperty()
+        {
+            _writer.Line();
+            _writer.WriteXmlDocumentationSummary($"Typed Resource Identifier for the container.");
+            _writer.LineRaw("// todo: hard coding ResourceGroupResourceIdentifier we don't know the exact ID type but we need it in implementations in CreateOrUpdate() etc.");
+            _writer.Line($"public new {typeof(ResourceGroupResourceIdentifier)} Id => base.Id as {typeof(ResourceGroupResourceIdentifier)};");
+        }
+
+        private void WriteResourceOperations()
+        {
+            _writer.Line();
+            _writer.LineRaw($"// Container level operations.");
+
+            // To generate resource operations, we need to find out the correct REST client methods to call.
+            // We must look for CreateOrUpdate by HTTP method because it may be named differently from `CreateOrUpdate`.
+            if (FindRestClientMethodByHttpMethod(RequestMethod.Put, out var restClientMethod))
+            {
+                WriteCreateOrUpdateVariants(restClientMethod);
+            }
+
+            // We must look for Get by method name because HTTP GET may map to >1 methods (get/list)
+            if (FindRestClientMethodByName(new string[] { "Get" }, out restClientMethod))
+            {
+                WriteGetVariants(restClientMethod);
+            }
+            else
+            {
+                WriteGetVariantsThatThrow();
```

Comment:
When the resource does not support GET, it should inherit `ContainerBase` not `ResourceCOntainerBase`